MatrixSDK/MXSession.m

```diff
@@ -5196,4 +5196,68 @@ - (NSString *)preferredSyncPresenceString
     return [MXTools presenceString:self.preferredSyncPresence];
 }
 
+#pragma mark - Homeserver OAuth 2.0 metadata
+
+- (bool)hasOAuth2APIEnabled
+{
+    return self.store.authMetadata != nil || self.store.homeserverWellknown.authentication != nil;
+}
+
+- (NSString *)accountManagementUri
+{
+    if (self.store.authMetadata)
+    {
+        return self.store.authMetadata.accountManagementUri;
+    }
+    
+    if (self.store.homeserverWellknown.authentication)
+    {
+        return self.store.homeserverWellknown.authentication.account;
+    }
+    
+    return nil;
+}
+
+-(NSURL * _Nullable) getLogoutDeviceURLFromID: (NSString * ) deviceID
+{
+    if (self.store.authMetadata)
+    {
+        return [self.store.authMetadata getLogoutDeviceURLFromID:deviceID];
+    }
+    
+    if (self.store.homeserverWellknown.authentication)
+    {
+        return [self.store.homeserverWellknown.authentication getLogoutDeviceURLFromID:deviceID];
+    }
+    
+    return nil;
+}
+
+
+- (MXAuthMetadata *)authMetadata
+{
+    return self.store.authMetadata;
+}
+
+- (MXHTTPOperation *)refreshAuthMetadata:(void (^)(MXAuthMetadata *))success
```

Comment:
I might be missing it, but I don't see this being called either in this PR, or in https://github.com/element-hq/element-ios/pull/7991.

I would assume we should drop it in here:

https://github.com/matrix-org/matrix-ios-sdk/blob/b0f967c9c884e1a24d037f516f40a5ffb6c29586/MatrixSDK/MXSession.m#L1046